server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/TraceServiceElasticsearch.java

```diff
@@ -132,25 +134,69 @@ public Trace getFragment(String tenantId, String id) {
         return ret;
     }
 
+    protected List<Trace> getFragmentsForTraceId(String tenantId, String traceId) {
+        List<Trace> ret = new ArrayList<>();
```

Comment:
I'm not sure if the `response.getHits()` has `size()` method, if so it would be good to use it in array list construction.